### ISSUE TITLE
Repeated read/processing of package.json

### ISSUE THREAD

**Issue by JasonKleban** [eslint/eslint#14187]
Repeated reads/parsing of package.json (and prettier.rc, .prettierignore) for every source file linted.

**Tell us about your environment**

Node version: v14.13.0
npm version: v6.14.4
Local ESLint version: v7.21.0 (Currently used)
Global ESLint version: Not found

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

`@typescript-eslint/parser`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    parser: "@typescript-eslint/parser", // Specifies the ESLint parser
    extends: [
        "plugin:react/recommended",
        "plugin:@typescript-eslint/recommended",
        "prettier/@typescript-eslint",
        "plugin:prettier/recommended",
        "plugin:json/recommended",
        "plugin:react-hooks/recommended",
    ],
    env: {
        node: true,
        jest: true,
    },
    plugins: ["react", "lodash", "react-hooks"],
    parserOptions: {
        ecmaFeatures: {
            jsx: true,
            experimentalObjectRestSpread: true,
            defaultParams: true,
            destructuring: true,
        },
    },
    ignorePatterns: [ "**/node_modules", "**/dist" ],
    rules: {
        "no-debugger": "error",
        "no-console": ["error", { allow: ["warn", "error"] }],
        "no-unused-vars": "warn",
        complexity: "off",
        "no-unneeded-ternary": "warn",
        "prefer-rest-params": "warn",
        "no-undef": "warn",
        "one-var": "off",
        "ban-ts-ignore": "off",
        "brace-style": "off",
        "valid-jsdoc": "off",
        "no-fallthrough": "off",
        "max-len": "off",
        radix: "off",
        eqeqeq: ["warn", "smart"],
        "new-cap": ["warn", { capIsNew: false }],
        "no-var": "off",
        "no-redeclare": "off",
        "prefer-const": "warn",
        "prefer-spread": "warn",
        "prettier/prettier": "warn",
        camelcase: "off",
        "no-lone-blocks": "off",
        "react/display-name": "warn",
        "react/jsx-key": "warn",
        "react/no-children-prop": "warn",
        "react/react-in-jsx-scope": "warn",
        "react/no-unknown-property": "warn",
        "react/jsx-no-bind": [1],
        "react/jsx-no-target-blank": "warn",
        "react/jsx-uses-react": "warn",
        "react/no-string-refs": "warn",
        "react/no-unescaped-entities": "warn",
        "react/no-find-dom-node": "warn",
        "react/no-direct-mutation-state": "warn",
        "react/jsx-uses-vars": "error",
        "react/jsx-no-undef": "warn",
        "react/prop-types": [0],
        "react-hooks/rules-of-hooks": "error",
        "react-hooks/exhaustive-deps": "warn",
        "@typescript-eslint/no-explicit-any": "warn",
        "@typescript-eslint/no-var-requires": "off",
        "@typescript-eslint/ban-ts-comment": "warn",
        "@typescript-eslint/explicit-function-return-type": "off",
        "@typescript-eslint/explicit-module-boundary-types": "off",
        "@typescript-eslint/camelcase": ["off", { properties: "always" }],
        "@typescript-eslint/explicit-member-accessibility": "off",
        "@typescript-eslint/interface-name-prefix": ["off", "always"],
        "@typescript-eslint/no-parameter-properties": "warn",
        "@typescript-eslint/no-use-before-define": "warn",
        "@typescript-eslint/ban-types": "off",
        "@typescript-eslint/no-inferrable-types": "warn",
        "@typescript-eslint/consistent-type-assertions": "warn",
        "@typescript-eslint/no-empty-function": "warn",
        "@typescript-eslint/no-empty-interface": "warn",
        "@typescript-eslint/prefer-namespace-keyword": "warn",
        "@typescript-eslint/no-this-alias": "warn",
        "@typescript-eslint/no-array-constructor": "warn",
        "prefer-destructuring": [
            "off",
            {
                object: true,
                array: false,
            },
        ],
        "sort-imports": [
            "error",
            {
                ignoreCase: true,
                ignoreDeclarationSort: true,
                ignoreMemberSort: true,
                memberSyntaxSortOrder: ["none", "all", "multiple", "single"],
            },
        ],
        "padding-line-between-statements": [
            "off",
            {
                blankLine: "always",
                prev: "*",
                next: "export",
            },
            {
                blankLine: "always",
                prev: "*",
                next: "function",
            },
            {
                blankLine: "always",
                prev: "*",
                next: "return",
            },
            {
                blankLine: "always",
                prev: "*",
                next: "switch",
            },
            {
                blankLine: "always",
                prev: "*",
                next: "if",
            },
            {
                blankLine: "always",
                prev: "*",
                next: "for",
            },
        ],
        "lodash/import-scope": [2, "method"],
        "no-submodule-imports": [
            2
        ]
    },
    settings: {
        react: {
            version: "detect", 
        },
    },
    globals: {
        document: true,
        google: true,
        window: true,
    }
};
```
</details>

**What did you expect to happen?**

Eslint should not have to re-read and re-parse, etc certain files for a single linting execution.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

![image](https://user-images.githubusercontent.com/17498172/110376547-5566ab00-8021-11eb-9d4e-63b066326ab6.png)

**Steps to reproduce this issue:**

1. Use Windows Process Monitor to capture all ReadFile operations from node.exe processes.
1. See that certain files are referenced repeatedly.
1. Note that in the Tools > File Summary, the counts of full file reads (and then, presumably, parsing and processing of that same content).

**Are you willing to submit a pull request to fix this bug?**

Probably this needs an expert to fix this.

**Comment by JasonKleban** [comment]
Actually I think this is a problem with `plugin:prettier/recommended` from `eslint-plugin-prettier`.  When I upgraded from "eslint-plugin-prettier": "3.1.4", to "eslint-plugin-prettier": "3.3.1", the package.json stuff stopped but .prettierignore remains.  only commenting out `extends: [ ... "plugin:prettier/recommended"` gets completely rid of the extra reads.

**Comment by nzakas** [comment]
Yeah, we only read package.json one time. There’s no need to for us to read it again so it’s most likely either a plugin you’re using or a parser you’re using.

**Comment by nzakas** [comment]
Closing, as this doesn’t appear to be an issue with ESLint itself.
